src/tr/tr.rs

```diff
@@ -47,24 +50,28 @@ fn delete(set: ExpandSet, complement: bool) {
         }
     };
 
-    let mut reader = BufReader::new(stdin());
+    stdin.lock();
 
-    while let Ok(length) = reader.read_to_string(&mut buf) {
+    while let Ok(length) = stdin.read_line(&mut buf) {
         if length == 0 { break }
-
-        let filtered = buf.chars()
-                          .filter(|c| { is_allowed(*c) })
-                          .collect::<String>();
-        safe_unwrap!(stdout.write_all(filtered.as_bytes()));
+        { // isolation to make borrow checker happy
+            let filtered = buf.chars().filter(|c| { is_allowed(*c) });
```

Comment:
The compiler won't let me simply drop the braces:

```
error[E0502]: cannot borrow `buf` as mutable because it is also borrowed as immutable
  --> tr.rs:62:9
   |
57 |         let filtered = buf.chars().filter(|c| { is_allowed(*c) });
   |                        --- immutable borrow occurs here
...
62 |         buf.clear();
   |         ^^^ mutable borrow occurs here
63 |     }
   |     - immutable borrow ends here
```

So unless you can suggest a way to rewrite the code wrapped in braces they'll have to stay.